htdocs/core/lib/security.lib.php

```diff
@@ -382,7 +382,7 @@ function checkUserAccessToObject($user, $featuresarray, $objectid=0, $tableandsh
 		$checkother = array('contact','agenda');	 // Test on entity and link to third party. Allowed if link is empty (Ex: contacts...).
 		$checkproject = array('projet','project'); // Test for project object
 		$checktask = array('projet_task');
-		$nocheck = array('barcode','stock');	// No test
+		$nocheck = array('barcode','stock','fournisseur');	// No test
```

Comment:
If you add this, test on permission on $featuresarray=('fournisseur','commande') is not done and users can always see all orders of all entities even if inside a non master entity.

Reply:
@eldy 
without "fournisseur" an error appears on the page "/product/provider.php" when the user is limited to seeing that the thirds which are attributed to him.

Param dbt_keyfield is required but not defined

if i add "dbt_keyfield" the user can not access to the page !

here "dbt_keyfield" is "fk_soc" (wrong solution)

$result=restrictedArea($user,'produit|service&fournisseur',$fieldvalue,'product&product','','fk_soc',$fieldtype);